crates/storage/src/lazy/lazy_array.rs

```diff
@@ -254,31 +214,28 @@ impl<'a, T> DoubleEndedIterator for EntriesIter<'a, T> {
 
 impl<'a, T> ExactSizeIterator for EntriesIter<'a, T> {}
 
-impl<T, N> EntryArray<T, N>
-where
-    N: LazyArrayLength<T>,
-{
+impl<T, const N: usize> EntryArray<T, N> {
     /// Creates a new entry array cache.
     pub fn new() -> Self {
-        Self {
-            entries: Default::default(),
-        }
+        let entries = iter::repeat_with(Default::default)
+            .take(N)
+            .collect::<Vec<CacheCell<Option<StorageEntry<T>>>>>()
+            .try_into();
+        let entries = match entries {
+            Ok(entries) => entries,
+            Err(_) => unreachable!("try_into must work"),
+        };
+        Self { entries }
```

Comment:
Whoat? Looks way more complicated than before and also seems to be allocating heap memory which is against the entire point of small vec and lazy array data structures. Please elaborate why.

Reply:
Yeah… I pushed a different approach now, that one requires `#![feature(array_map)]` though 🙈.

Reply:
Nightly Rust features are a show stopper for us. Please look if the `array_init` crate can help you here. We are already depending on it anyway.